### DISCUSSION TITLE
Colvars Errors to restart

### DISCUSSION THREAD

**Discussion by kail0221** [Colvars/colvars#742] [Q&A] [answered]
Dear all,

I encountered an issue while attempting to restart the crashed simulation. The problem is that during the simulation, the dihedral saved in `colvars.traj ` and colvars.state differs from the one calculated directly using a file like `restart.coor and .dcd file`. This discrepancy prevents me from restarting the interrupted simulation, as I receive the following error message:

"The calculated value of colvar 'dihedral1': -69.1871 differs greatly from the value last read from the state file: -51.4628."

When I manually visualize the restart file and calculate the dihedral1 , it is indeed -69.19 and not -51.4628 as recorded in the `colvars.state` and `colvars.traj` file.

I have tried using `WrapAll off` and `forceNoPBC`, but neither option has resolved the issue. I' ve attached the relevant files, including the error log file (`fe7.log`), for your reference.

[Abl-Kinase.zip](https://github.com/user-attachments/files/17528857/Abl-Kinase.zip)

**Comment by giacomofiorin** [comment]
Hi @kail0221!

From a quick look at the fe6.log file it looks like the issue may be having set `firstTimeStep` as a number that is not a multiple of your restart frequency. At the end of the file, I cannot confirm that the Colvars state file and the NAMD coor file are being written at the same simulation step.

**Reply by kail0221** [reply]
Hi @giacomofiorin ,
Thank you for your response.
Yes, it’s true that the simulation crashed at a point that wasn’t a multiple of my save frequency. However, when restarting, I have consistently used the restart files（`.restart.coor and .restart.colvars.state`）, which are all saved at the same step—specifically, at step 37,710,000.

**Reply by giacomofiorin** [reply, marked as answer]
Let's take a look at the end of `fe6.log`:

### Step 37710000: restarts are saved
Here, Colvars saves the state file at step 37710000, followed by NAMD a few lines below. (Colvars is called relatively early in a step cycle, whereas NAMD does its output toward the end of it.)
```
colvars: Synchronizing (emptying the buffer of) trajectory file "output2/win_15_-135_6.colvars.traj".
colvars: Saving collective variables state to "output2/win_15_-135_6.restart.colvars.state".
PRESSURE: 37710000 -214.403 -104.676 -136.595 -104.674 -330.659 81.0438 -136.594 81.045 -138.079
GPRESSURE: 37710000 -240.726 -40.7335 -79.8912 -96.6888 -243.775 76.1312 -103.967 101.778 -128.377
PRESSAVG: 37710000 -27.1951 -24.8215 -4.86062 -24.8215 -4.00431 -21.5267 -4.86062 -21.5267 -18.9668
GPRESSAVG: 37710000 -18.1205 -6.68883 -8.28621 -28.4431 18.7514 -9.09616 7.42104 -45.7356 5.15319
ENERGY: 37710000       848.1808      2272.4089      2327.3668       132.2062        -219680.7996     18886.9425         0.0000       -12.09>

WRITING EXTENDED SYSTEM TO RESTART FILE AT STEP 37710000
WRITING COORDINATES TO DCD FILE output2/win_15_-135_6.dcd AT STEP 37710000
WRITING COORDINATES TO RESTART FILE AT STEP 37710000
FINISHED WRITING RESTART COORDINATES
WRITING VELOCITIES TO RESTART FILE AT STEP 37710000
FINISHED WRITING RESTART VELOCITIES
```

### Step 37712000: no output
Then comes step 37712000, where neither Colvars nor NAMD save anything.
```
PRESSURE: 37712000 82.364 -9.51279 154.178 -9.51283 43.5627 56.6738 154.18 56.6738 44.7962
GPRESSURE: 37712000 133.385 49.0085 102.134 -44.612 34.2263 102.349 130.495 123.537 32.429
PRESSAVG: 37712000 -13.8769 -21.3261 1.01438 -21.3261 7.47397 -19.7283 1.01438 -19.7283 -20.2074
GPRESSAVG: 37712000 -0.543732 -1.46703 -2.34285 -27.5176 26.3055 -7.80767 12.0223 -38.3987 2.43488
ENERGY: 37712000       825.6338      2270.8886      2353.5954       120.5886        -219918.4522     19422.5601         0.0000       -12.30>
```

### Step 37713140: end of the simulation
Then comes the end of the simulation, which is step 37713140 based on your earlier definition of `firstTimeStep`. Colvars saves both the restart file one more time and the final output file. We need to save both, because we cannot know which one is defined depending on the specifics of the NAMD script.

NAMD also saves its final output file, which looking at your log file should be named `output2/win_15_-135_6.coor`. **This is the file that you should use in combination with `output2/win_15_-135_6.colvars.state`.** I am reasonably confident that these will be in sync.

Note that the last three lines below indicate the _normal conclusion of a NAMD run_: there was no crash here, and the restart files (NAMD's or Colvars') are not needed.
```
colvars: Synchronizing (emptying the buffer of) trajectory file "output2/win_15_-135_6.colvars.traj".
colvars: Saving collective variables state to "output2/win_15_-135_6.restart.colvars.state".
colvars: Saving collective variables state to "output2/win_15_-135_6.colvars.state".
WRITING EXTENDED SYSTEM TO OUTPUT FILE AT STEP 37713140
CLOSING EXTENDED SYSTEM TRAJECTORY FILE
WRITING COORDINATES TO OUTPUT FILE AT STEP 37713140
CLOSING COORDINATE DCD FILE output2/win_15_-135_6.dcd
WRITING VELOCITIES TO OUTPUT FILE AT STEP 37713140
====================================================

WallClock: 28783.873047  CPUTime: 27459.818359  Memory: 0.000000 MB
[Partition 0][Node 0] End of program
```

### General observations
- Here, your restart frequency combined with your wall time per step means that you are writing all information again every 13 seconds: for most people, this time interval should be a few minutes or longer.
- Although Colvars' approach is usually robust, edge cases like this make it less so. Please always keep the number of steps in a run and the first time step both a multiple of your restart frequency.
- In the future, if NAMD were to start using a single restart file for all data, we'd be happy to incorporate Colvars data in it. This is what we are doing in LAMMPS and GROMACS.

**Reply by kail0221** [reply]
It is indeed what you said, I understood it completely and thank you very much for helping me explain the log file line by line!
